lib/ansible/plugins/action/copy.py

```diff
@@ -199,6 +199,15 @@ def _recurse(topdir, rel_offset, parent_dirs, rel_base=u''):
     return r_files
 
 
+def combine_result_diffs(dest, source):
+    if not source.get("diff", None):
+        return
+    if isinstance(source["diff"], list):
+        dest["diff"] += source["diff"]
+    else:
+        dest["diff"].append(source["diff"])
+
+
```

Comment:
The bug in the module does not require any modifications to the action plugin. Please remove the new function in preparation of changing the copy module. Your code change looked fine to me before, just use `+=`/`.extend()` inline.

Reply:
ok, I put it back, but I don't understand. The original logic was to concatenate `copy`'s `result.diff`, but append `file`'s `result.diff`. This only works if `file`'s `result.diff` is a dictionary, not a list. It seems fragile to me to make that assumption about any module's result, hence the function.